examples/tests/valid/json.w

```diff
@@ -21,6 +21,10 @@ assert(json_mut_obj.get("hello") == message);
 // Assignment from natives
 let some_number: num = 999;
 
+let jj = Json some_number;
+let jj1 = Json {foo: some_number};
+let jj2 = Json [ some_number, {bar: some_number} ];
```

Comment:
What happens if someone uses some other kind of expression? e.g.

```
let jj3 = Json returns_number();
let jj4 = Json foo.bar;
```

Part of me feels like either all of these should work or none of these should work (but it's definitely not a hill I'm going to die on)

Reply:
Yea im not sure where the line is, if there is one. Maybe we do just allow it all, and change the json element to be any valid expression.

I dont have any super strong opinions here. Maybe just go ahead and open up the flood gates and see where that takes us?

Reply:
I'm for changing this to any valid expression, but forcing the type checker to validate the expression's type evaluates to either a Json or a Json-convertable value  (a primitive).

Reply:
I'm with @yoav-steinberg's suggestion